### ISSUE TITLE
Casper Wallet Adapter

### ISSUE THREAD

**Issue by berzanorg** [casper-ecosystem/developer-rewards#31]
### Reward Size in USD

700 USD

### Reward category

SDK

### Description

Create an NPM package for connecting to different wallets in the Casper ecosystem. A standardized wallet adapter for Casper is going to enable dApp developers to start building quickly without the need to reinvent the wheel. It is going to be developed using React 18. The project structure will be kept as simple as possible to allow new contributors to freely understand the project. Having a lightweight and performant package is the goal. Lastly, a good README.md file is going to show how it is used. 
Currently I've found three different wallets (except Casper Signer as it going to be deprecated):
 - Casper Wallet
 - Gosuto Wallet
 - CasperDash Wallet
Those wallets are going to be supported. 
The package will provide an easy-to-use interface for everything a dApp needs to interact with wallets. 

### Acceptance Criteria

1. Find new wallets to support if there are missing ones.
2. Read documentations of each wallet.
3. Build Casper Wallet Adapter NPM package.
4. Make a good README.md file that shows how the package is going to be used.

**Comment by NicolasZoellner** [comment]
Thank you, @BerzanXYZ, for your proposal. We will evaluate it during the next evaluation meeting scheduled for Tuesday, September 19th.
Best regards
Nicolas

**Comment by gaupoit** [comment]
We already developed the React Hooks library (https://github.com/CasperDash/useWallet) containing everything developers need to start connecting their dApps with Casper Wallets:

- CasperDash Wallet
- Casper Wallet
- Ledger

We might develop from this for NPM Package and support other JS Framework such as VueJS.

**Comment by berzanorg** [comment]
Hmm. What about starting from scratch while also supporting all the popular frameworks at once, React, Vue and Svelte. Maybe a tool like `create-vite` can be developed on top of it for providing a template for dApp developers to start with.

**Comment by NicolasZoellner** [comment]
Hello @BerzanXYZ,

Thank you for your contribution. We assessed your proposal during the technical evaluation meeting on September 19, 2023.

There are already existing solutions, as @gaupoit mentioned earlier:
https://github.com/CasperDash/useWallet

Furthermore, there is already a solution in place from Make as well:
https://github.com/make-software/casper-wallet

All of these are open-source and ready to use.

Therefore, we do not support the rebuilding of existing infrastructure, and we have concluded that your proposal is rejected.

We appreciate your contribution.

Best regards,
Nicolas Zöllner
